src/Sulu/Component/Content/Repository/ContentRepository.php

```diff
@@ -0,0 +1,636 @@
+<?php
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Component\Content\Repository;
+
+use Jackalope\Query\QOM\PropertyValue;
+use Jackalope\Query\Row;
+use PHPCR\ItemNotFoundException;
+use PHPCR\Query\QOM\QueryObjectModelFactoryInterface;
+use PHPCR\SessionInterface;
+use PHPCR\Util\QOM\QueryBuilder;
+use Sulu\Component\Content\Compat\LocalizationFinder;
+use Sulu\Component\Content\Compat\StructureType;
+use Sulu\Component\Content\Document\RedirectType;
+use Sulu\Component\Content\Repository\Mapping\MappingInterface;
+use Sulu\Component\DocumentManager\PropertyEncoder;
+use Sulu\Component\Localization\Localization;
+use Sulu\Component\PHPCR\SessionManager\SessionManagerInterface;
+use Sulu\Component\Security\Authentication\UserInterface;
+use Sulu\Component\Util\SuluNodeHelper;
+use Sulu\Component\Webspace\Manager\WebspaceManagerInterface;
+
+/**
+ * Content repository which query content with sql2 statements.
+ */
+class ContentRepository implements ContentRepositoryInterface
+{
+    // TODO bad name they should not be handled by redirects and shadow
+    private static $nonFallbackProperties = [
+        'uuid',
+        'state',
+        'order',
+        'created',
+        'creator',
+        'changed',
+        'changer',
+        'shadowOn',
+        'shadowBase',
+        'url', // TODO non fix name in templates
```

Comment:
hmm it would be a bc break when we say that the URL has to be named as URL .... and this class currently dont handles the URL correctly in case of internal/external link ... so i would let it in here and fix it in the next iteration!